src/ImageSharp/Common/Helpers/DenseMatrixUtils.cs

```diff
@@ -247,6 +248,91 @@ internal static class DenseMatrixUtils
+        /// <param name="maxColumn">The maximum working area column.</param>
+        public static void Convolve1D<TPixel>(
+            Span<Complex64> kernel,
+            Buffer2D<TPixel> sourcePixels,
+            Span<ComplexVector4> targetRow,
+            int row,
+            int column,
+            int minRow,
+            int maxRow,
+            int minColumn,
+            int maxColumn)
+            where TPixel : struct, IPixel<TPixel>
+        {
+            ComplexVector4 vector = default;
+            int kernelLength = kernel.Length;
+            int radiusY = kernelLength >> 1;
+            int sourceOffsetColumnBase = column + minColumn;
+            ref Complex64 baseRef = ref MemoryMarshal.GetReference(kernel);
+
+            for (int i = 0; i < kernelLength; i++)
+            {
+                int offsetY = (row + i - radiusY).Clamp(minRow, maxRow);
+                int offsetX = sourceOffsetColumnBase.Clamp(minColumn, maxColumn);
+                Span<TPixel> sourceRowSpan = sourcePixels.GetRowSpan(offsetY);
+                var currentColor = sourceRowSpan[offsetX].ToVector4();
+
+                vector.Sum(Unsafe.Add(ref baseRef, i) * currentColor);
+            }
+
+            targetRow[column] = vector;
+        }
+
+        /// <summary>
+        /// Computes the sum of vectors in <paramref name="targetRow"/> weighted by the kernel weight values.
+        /// </summary>
+        /// <param name="kernel">The 1D convolution kernel.</param>
+        /// <param name="sourceValues">The source frame.</param>
+        /// <param name="targetRow">The target row.</param>
+        /// <param name="row">The current row.</param>
+        /// <param name="column">The current column.</param>
+        /// <param name="minRow">The minimum working area row.</param>
+        /// <param name="maxRow">The maximum working area row.</param>
+        /// <param name="minColumn">The minimum working area column.</param>
+        /// <param name="maxColumn">The maximum working area column.</param>
+        public static void Convolve1D(
+            Span<Complex64> kernel,
+            Buffer2D<ComplexVector4> sourceValues,
+            Span<ComplexVector4> targetRow,
+            int row,
+            int column,
+            int minRow,
+            int maxRow,
+            int minColumn,
+            int maxColumn)
+        {
+            ComplexVector4 vector = default;
+            int kernelLength = kernel.Length;
+            int radiusX = kernelLength >> 1;
+            int sourceOffsetColumnBase = column + minColumn;
+
+            int offsetY = row.Clamp(minRow, maxRow);
+            ref ComplexVector4 sourceRef = ref MemoryMarshal.GetReference(sourceValues.GetRowSpan(offsetY));
+            ref Complex64 baseRef = ref MemoryMarshal.GetReference(kernel);
+
+            for (int x = 0; x < kernelLength; x++)
+            {
+                int offsetX = (sourceOffsetColumnBase + x - radiusX).Clamp(minColumn, maxColumn);
```

Comment:
How does this convolution work with semi-opaque pixels? We use alpha premultiplication for other blur algorithms to ensure color output is correct.

Reply:
So, just to follow up on our conversation on gitter.
In case you want to actually check the code, the premultiplication pass is done right before doing the initial gamma exposure correction, here: https://github.com/Sergio0694/ImageSharp/blob/8e9986e058b07d44a5ddf9990931ad335ced35a9/src/ImageSharp/Processing/Processors/Convolution/BokehBlurProcessor%7BTPixel%7D.cs#L470.

Then the various steps to apply the effect are performed, and the pixels are finally unpremultiplied after the inverse gamma exposure step, here: https://github.com/Sergio0694/ImageSharp/blob/8e9986e058b07d44a5ddf9990931ad335ced35a9/src/ImageSharp/Processing/Processors/Convolution/BokehBlurProcessor%7BTPixel%7D.cs#L533.